File: Source/OldCode/Archive/2006 Phasor Protocols/PhasorProtocols/BpaPdcStream/AnalogValue.vb
'*******************************************************************************************************
'  AnalogValue.vb - PDCstream Analog value
'  Copyright © 2005 - TVA, all rights reserved - Gbtc
'
'  Build Environment: VB.NET, Visual Studio 2005
'  Primary Developer: J. Ritchie Carroll, Operations Data Architecture [TVA]
'      Office: COO - TRNS/PWR ELEC SYS O, CHATTANOOGA, TN - MR 2W-C
'       Phone: 423/751-2827
'       Email: jrcarrol@tva.gov
'
'  Code Modification History:
'  -----------------------------------------------------------------------------------------------------
'  11/12/2004 - J. Ritchie Carroll
'       Initial version of source generated
'
'*******************************************************************************************************

Imports System.Runtime.Serialization

Namespace BpaPdcStream

    ''' <summary>
    ''' BPA PDCstream Analog Value Class
    ''' </summary>
    <CLSCompliant(False), Serializable()> _
    Public Class AnalogValue

        Inherits AnalogValueBase

        Protected Sub New()
        End Sub

        Protected Sub New(ByVal info As SerializationInfo, ByVal context As StreamingContext)

            MyBase.New(info, context)

        End Sub

        Public Sub New(ByVal parent As IDataCell, ByVal analogDefinition As IAnalogDefinition, ByVal value As Single)

            MyBase.New(parent, analogDefinition, value)

        End Sub

        Public Sub New(ByVal parent As IDataCell, ByVal analogDefinition As IAnalogDefinition, ByVal unscaledValue As Int16)

            MyBase.New(parent, analogDefinition, unscaledValue)

        End Sub

        Public Sub New(ByVal parent As IDataCell, ByVal analogDefinition As IAnalogDefinition, ByVal binaryImage As Byte(), ByVal startIndex As Int32)

            MyBase.New(parent, analogDefinition, binaryImage, startIndex)

        End Sub

        Public Sub New(ByVal parent As IDataCell, ByVal analogDefinition As IAnalogDefinition, ByVal analogValue As IAnalogValue)

            MyBase.New(parent, analogDefinition, analogValue)

        End Sub

        Friend Shared Function CreateNewAnalogValue(ByVal parent As IDataCell, ByVal definition As IAnalogDefinition, ByVal binaryImage As Byte(), ByVal startIndex As Int32) As IAnalogValue

            Return New AnalogValue(parent, definition, binaryImage, startIndex)

        End Function

        Public Overrides ReadOnly Property DerivedType() As System.Type
            Get
                Return Me.GetType
            End Get
        End Property

        Public Shadows ReadOnly Property Parent() As DataCell
            Get
                Return MyBase.Parent
            End Get
        End Property

        Public Shadows Property Definition() As AnalogDefinition
            Get
                Return MyBase.Definition
            End Get
            Set(ByVal value As AnalogDefinition)
                MyBase.Definition = value
            End Set
        End Property

    End Class

End Namespace

File: Source/OldCode/Archive/2007 Phasor Protocols/PhasorProtocols/BpaPdcStream/AnalogValue.cs
using System.Diagnostics;
using System;
//using PCS.Common;
using System.Collections;
using PCS.Interop;
using Microsoft.VisualBasic;
using PCS;
using System.Collections.Generic;
//using PCS.Interop.Bit;
using System.Linq;
using System.Runtime.Serialization;

//*******************************************************************************************************
//  AnalogValue.vb - PDCstream Analog value
//  Copyright © 2008 - TVA, all rights reserved - Gbtc
//
//  Build Environment: VB.NET, Visual Studio 2008
//  Primary Developer: J. Ritchie Carroll, Operations Data Architecture [TVA]
//      Office: COO - TRNS/PWR ELEC SYS O, CHATTANOOGA, TN - MR 2W-C
//       Phone: 423/751-2827
//       Email: jrcarrol@tva.gov
//
//  Code Modification History:
//  -----------------------------------------------------------------------------------------------------
//  11/12/2004 - J. Ritchie Carroll
//       Initial version of source generated
//
//*******************************************************************************************************


namespace PCS.PhasorProtocols
{
    namespace BpaPdcStream
    {

        /// <summary>
        /// BPA PDCstream Analog Value Class
        /// </summary>
        [CLSCompliant(false), Serializable()]
        public class AnalogValue : AnalogValueBase
        {



            protected AnalogValue()
            {
            }

            protected AnalogValue(SerializationInfo info, StreamingContext context)
                : base(info, context)
            {


            }

            public AnalogValue(IDataCell parent, IAnalogDefinition analogDefinition, float value)
                : base(parent, analogDefinition, value)
            {


            }

            public AnalogValue(IDataCell parent, IAnalogDefinition analogDefinition, short unscaledValue)
                : base(parent, analogDefinition, unscaledValue)
            {


            }

            public AnalogValue(IDataCell parent, IAnalogDefinition analogDefinition, byte[] binaryImage, int startIndex)
                : base(parent, analogDefinition, binaryImage, startIndex)
            {


            }

            public AnalogValue(IDataCell parent, IAnalogDefinition analogDefinition, IAnalogValue analogValue)
                : base(parent, analogDefinition, analogValue)
            {


            }

            internal static IAnalogValue CreateNewAnalogValue(IDataCell parent, IAnalogDefinition definition, byte[] binaryImage, int startIndex)
            {

                return new AnalogValue(parent, definition, binaryImage, startIndex);

            }

            public override System.Type DerivedType
            {
                get
                {
                    return this.GetType();
                }
            }

            public new DataCell Parent
            {
                get
                {
                    return (DataCell)base.Parent;
                }
            }

            public new AnalogDefinition Definition
            {
                get
                {
                    return (AnalogDefinition)base.Definition;
                }
                set
                {
                    base.Definition = value;
                }
            }

        }

    }
}
